packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -228,7 +255,12 @@ class PubspecYaml {
         final PubspecHeader header = PubspecHeader.parse(line); // See if it's a header.
         if (header != null) { // It is!
           section = header.section; // The parser determined what kind of section it is.
-          if (section == Section.dependencies) {
+          if (section == Section.header) {
+            if (header.name == 'name')
+              packageName = header.value;
+            else if (header.name == 'version')
+              packageVersion = header.value;
+          } else if (section == Section.dependencies) {
```

Comment:
looks like you ended up not using this?

Reply:
Oh, right. Cleaning up.

Reply:
Oh, wait, sorry, this _is_ used. See line 157.